### PR TITLE
Improve discovery/upgrade messages

### LINKED ISSUE #11992: Errors updating azdata
2 reported issues : 

1. This error is appearing - looks like some path isn't quoted properly

```
Error updating azdata : Unexpected exit code from command : '""C:\Program' is not recognized as an internal or external command,
operable program or batch file. (1)
```

2. `Error updating azdata: not found: azdata.cmd` on startup when azdata isn't installed. We shouldn't be displaying that error during update when it isn't user-prompted (instead just fail silently)

### PR DESCRIPTION
**Pull request by ranasaria**
This change fixes an issue where spurious upgrade messages were reported when no azdata was found to be installed. It also adds/fixes other messages to make information in output channel clearer.

This PR fixes #11992

### CODE REVIEW

File: extensions/azdata/src/extension.ts [hunk 1]
```diff
@@ -13,11 +13,18 @@ let localAzdata: IAzdataTool | undefined = undefined;
...
 	checkAndUpgradeAzdata(localAzdata)
-		.then(async () => {
-			localAzdata = await findAzdata(); // now again find and return the currently installed azdata
+		.then(async upgradePerformed => {
+			try {
+				if (upgradePerformed) { // If upgrade was performed then find the new azdata and save the new azdata
+					localAzdata = await findAzdata();
+				}
+			} catch (_err) {
+				//Do nothing, it is expected that if user did not accept upgrade of azdata then findAzdata will still fail throwing an error.
```
Comment (Charles-Gagnon) [inline extensions/azdata/src/extension.ts:24]: Why is this necessary? You check if the upgrade was performed so I would expect this to work in most cases.

Anyways let's at least log to the console here for debugging.

File: extensions/azdata/src/localizedConstants.ts [hunk 2]
```diff
@@ -23,14 +23,16 @@ export const stdoutOutput = (stdout: string): string => localize('azdata.stdoutO
...
 export const checkingLatestAzdataVersion = localize('azdata.checkingLatestAzdataVersion', "Checking for latest version of azdata");
 export const gettingTextContentsOfUrl = (url: string): string => localize('azdata.gettingTextContentsOfUrl', "Getting text contents of resource at URL {0}", url);
-export const foundAzdataVersionToUpgradeTo = (version: string): string => localize('azdata.versionForUpgrade', "Found version {0} that azdata-cli can be upgraded to.", version);
-export const promptForAzdataUpgrade = (version: string): string => localize('azdata.promptForAzdataUpgrade', "An updated version of azdata ( {0} ) is available, do you wish to install it now?", version);
+export const latestAzdataVersionAvailable = (version: string): string => localize('azdata.latestAzdataVersionAvailable', "Latest available azdata version: {0}.", version);
+export const currentlyInstalledVersionIsLatest = (currentVersion: string): string => localize('azdata.currentlyInstalledVersionIsLatest', "Currently installed version of azdata:{0} is same or newer than any other version avalailble", currentVersion);
```
Comment (Charles-Gagnon) [inline extensions/azdata/src/localizedConstants.ts:27]: should probably be a space in `azdata:{0}`

Comment (Charles-Gagnon) [inline extensions/azdata/src/localizedConstants.ts:27]: typo : `other version available`

File: extensions/azdata/src/extension.ts (hunk 1 above)
Comment (ranasaria) [reply extensions/azdata/src/extension.ts:24]: Will conosle.log the error.
